specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/networkManagerActiveConfiguration.json

```diff
@@ -0,0 +1,240 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "NetworkManagerActiveConfiguration",
+    "description": "The Microsoft Azure Network Manager Activity Configuration API provides a RESTful set of web services that interact with Microsoft Azure Network Manager service to manage your network resources.",
+    "version": "2019-11-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/networkManagers/{networkManagerName}/activeConfiguration/{locationName}": {
```

Comment:
>locationName [](start = 158, length = 12)

Did you guys look into location based routing ? (check arm wiki for 'routing types')

Reply:
**LocationBased**
This is allowed only on nested or proxy resource types. This routing type finds the registered endpoint by using the parent resource name as the location key. The advantage of this type is any client error in constructing the request URI (e.g.: "west us" or "us west" instead of "westus") will be flagged as a 400 with a clear and specific error message. This routing type is recommended for any location specific APIs that have location in the URL (e.g. Quota/metadata/capabilities)

E.g.: Requests to "providers/Microsoft.Compute/locations/westus/Quota" will be routed to whatever endpoint is registered under type Quota for location "westus".

---
In reply to: [388419103](https://github.com/Azure/azure-rest-api-specs/pull/8425#discussion_r388419103) [](ancestors = 388419103)